src/matrix/room/timeline/persistence/GapWriter.js

```diff
@@ -252,3 +254,268 @@ export class GapWriter {
         return {entries, updatedEntries, fragments};
     }
 }
+
+import {FragmentIdComparer} from "../FragmentIdComparer.js";
+import {RelationWriter} from "./RelationWriter.js";
+import {createMockStorage} from "../../../../mocks/Storage.js";
+import {FragmentBoundaryEntry} from "../entries/FragmentBoundaryEntry.js";
+import {createEvent, withTextBody, withContent, withSender} from "../../../../mocks/event.js";
+import {NullLogger} from "../../../../logging/NullLogger.js";
+
+export function tests() {
+    const alice = "alice@hs.tdl";
+    const bob = "bob@hs.tdl";
+    const roomId = "!room:hs.tdl";
+    const startToken = "begin_token";
```

Comment:
I wonder if using static tokens here might hide bugs if we don't persist the right token, or don't overwrite it. How about using the index as the token?